src/guide/arrays.md

```diff
@@ -0,0 +1,103 @@
+---
+title: Arrays
+---
+
+An alternative to lists, arrays are fixed-length containers for ordered data. Individual elements can be swapped out for other elements, and individual elements can be retrieved quickly.
+
+## Creating and Using Arrays
+
+Like lists, arrays always contain elements of the same type, and attempting to mix element types will result in a compilation error.
+
+```grain
+let empty = [>]
+let numbers = [> 1, 2, 3]
+let strings = [> "foo", "bar", "baz"]
+```
+
+Unlike lists, we can easily access elements at any (zero-based) index:
+
+```grain
+let strings = [> "foo", "bar", "baz"]
+
+print(strings[0]) // "foo"
+print(strings[1]) // "bar"
+print(strings[2]) // "baz"
+```
+
+We can also use negative indexes to access elements from the end of the array:
+
+```grain
+let strings = [> "foo", "bar", "baz"]
+
+print(strings[-1]) // "baz"
+print(strings[-2]) // "bar"
+print(strings[-3]) // "foo"
+```
+
+If we try to access an element beyond the length of the array, we'll get an `IndexOutOfBounds` error.
+
+## Updaing Arrays
+
+One of the major benefits arrays have is the ability to change the values it contains. We can update an array's values like so:
+
+```grain
+let strings = [> "foo", "bar", "baz"]
+
+print(strings) // [> "foo", "bar", "baz"]
+
+strings[1] = "qux"
+
+print(strings) // [> "foo", "qux", "baz"]
+```
+
+In some cases, this could allow us to write programs that are more efficient than if we used a list.
+
+However, the size of an array is fixed. To add additonal items to an array, we must append them together, which would create a brand new, third array:
+
+```grain
+import Array from "array"
+
+let one = [> 1]
+let twoThree = [> 2, 3]
+let oneTwoThree = Array.append(one, twoThree)
+
+print(oneTwoThree) // [> 1, 2, 3]
+```
+
+Since `oneTwoThree` is a brand new array, updating values in `one` or `twoThree` doesn't affect the values in `oneTwoThree`.
+
+For long arrays, even when adding just one element, this could be a fairly expensive operation. For programs that need to do this kind of operation, lists may be a better choice. We discuss this in more detail in the section below.
+
+To learn more about what's available in the Array standard library, check out the [array standard library documentation](https://grain-lang.org/docs/stdlib/array).
+
+## Lists vs. Arrays
+
+Lists and arrays are similar constructs, but each has its own pros and cons. It's up to you to choose which is right for your use case!
+
+Lists are excellent because they're
```

Comment:
```suggestion
Lists are excellent at
```

Reply:
I think`there` was a better transition here because, it goes on to list the properties rather than use cases. `at` indicates listing specific examples. It might make sense to say something like ```
Lists are useful over arrays because they are immutable, and their data structure makes resizing them more efficient, they are easier to work with in grain and less susceptible to bugs because it is a lot harder to get runtime errors. The downfalls of lists compared to arrays are that they are unable to modified in place, and their data structure makes accessing specific elements and determining the length less efficient. Lists are best used when the size does not matter and you are constantly appending or removing elements.
``` The downfall of that sentence structure is you lose the nice list though a better option over a list might be having the two paragraphs with their differences and then using a table to compare the features.